components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -17761,6 +17762,736 @@ public void deleteTenantTheme(int tenantId) throws APIManagementException {
+                    removeOrg(connection, organization);
+                }
+            }
+        }
+    }
+
+    /**
+     * Remove organization.
+     *
+     * @param connection   DB connection.
+     * @param organization Organization name.
+     */
+    private void removeOrg(Connection connection, String organization) throws SQLException {
+        String query = SQLConstants.DevPortalContentConstants.DELETE_ORG_ID;
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, organization);
+            preparedStatement.executeUpdate();
+        }
+    }
+
+    /**
+     * Imports a drafted api theme for the given organization and API ID.
+     *
+     * @param organization Organization name.
+     * @param themeContent Theme content as InputStream.
+     * @param apiId        API Identifier.
+     * @throws APIManagementException If a database error occurs.
+     */
+    public void importDraftedApiTheme(String organization, InputStream themeContent, String apiId)
+            throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+            try {
+                if (isApiAndOrganizationExist(connection, organization, apiId)) {
+                    String existingDraftedArtifact = getDraftedArtifactForApi(connection, organization, apiId);
+                    if (existingDraftedArtifact != null) {
+                        removeArtifact(connection, existingDraftedArtifact);
+                    }
+                    String newUUID = addArtifact(connection, themeContent, DevPortalConstants.DRAFTED_API_THEME);
+                    updateDraftedArtifactForApi(connection, organization, newUUID, apiId);
+                } else {
+                    String newUUID = addArtifact(connection, themeContent, DevPortalConstants.DRAFTED_API_THEME);
+                    insertNewApiWithDraftedArtifact(connection, organization, newUUID, apiId);
+                }
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
+                handleException("Failed to import drafted API theme for organization " + organization, e);
+            }
+        } catch (SQLException e) {
+            handleException("Database connection error while importing drafted API theme for organization " + organization, e);
+        }
+    }
+
+    /**
+     * Updates the api theme status as published or unpublished.
+     *
+     * @param organization Organization name.
+     * @param action       Action to perform ("PUBLISH" or "UNPUBLISH").
+     * @param apiId        API Identifier.
+     * @throws APIManagementException If a database error occurs.
+     */
+    public void updateApiThemeStatusAsPublishedOrUnpublished(String organization, String action, String apiId)
```

Comment:
rename the method as updateAPIThemeStatus

Reply:
Changed through https://github.com/wso2/carbon-apimgt/pull/12874/commits/439ab2cdf303bea83a11ef850970dd032e5c9f00